### ISSUE TITLE
[google_sign_in_ios] Update iOS SDK to 7.1.0 (was: Failing check_podspecs.)

### ISSUE THREAD

**Issue by ditman** [flutter/flutter#145866]
The recent update of the GoogleSignIn iOS SDK to v7.1.0 seems to have broken our CI in 2 ways:

1. **Mac_x64 check_podspecs** - [Build failure](https://ci.chromium.org/ui/p/flutter/builders/prod/Mac_x64%20check_podspecs/1573/overview).
2. **Mac_arm64 ios_platform_tests_shard_1 master** - [Build failure](https://ci.chromium.org/ui/p/flutter/builders/prod/Mac_arm64%20ios_platform_tests_shard_1%20master/1807/overview).

As a mitigation, the version of the SDK has been pinned to 7.0.0, here:

* https://github.com/flutter/packages/pull/6430

But the pin needs to be cleaned up.

----

_Earlier description_

The "Mac_x64 check_podspecs" seems to be consistently failing in flutter/packages; there's a warning while checking google_sign_in_os:

```
 - WARN  | swift: The validator used Swift `4.0` by default because no Swift
version was specified. To specify a Swift version during validation, add the
`swift_versions` attribute in your podspec. Note that usage of a `.swift-version`
file is now deprecated.
```

([Example of the Failure.](https://ci.chromium.org/ui/p/flutter/builders/try/Mac_x64%20check_podspecs/11731/overview))

~~In the meantime, `--allow-warnings` has been added to `podspec_check_command.dart` to unblock the tree. (Please, revert https://github.com/flutter/packages/pull/6416 as part of this fix!)~~ _Reverted, as the change exposed another issue, mentioned below!_

**Comment by ditman** [comment]
Here's a full run of the tool with the warnings allowed:

* [Build](https://ci.chromium.org/ui/p/flutter/builders/prod/Mac_x64%20check_podspecs/1570/overview) | [stdout](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8752259065149035441/+/u/Run_package_tests/validate_iOS_and_macOS_podspecs/stdout)

**Comment by ditman** [comment]
Skipping the podspec validation now exposes another error in `extension_google_sign_in_as_googleapis_auth ` (which depends on `google_sign_in`):

```
...
    -> Installing google_sign_in_ios (0.0.1)
      - Running pre install hooks
    [!] The following Swift pods cannot yet be integrated as static libraries:

    The Swift pod `GTMAppAuth` depends upon `GTMSessionFetcher` and `AppAuth`, which do not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.

    /Volumes/Work/s/w/ir/cache/ruby/.cipd/pkgs/0/pZTfjoFUwZNnilFjCkjEo8rdpdqPNerhy3XDi6ly8kAC/bin/darwin_ruby/lib/ruby/gems/3.1.0/gems/cocoapods-1.13.0/lib/cocoapods/installer/xcode/target_validator.rb:151:in `verify_swift_pods_have_module_dependencies'
    /Volumes/Work/s/w/ir/cache/ruby/.cipd/pkgs/0/pZTfjoFUwZNnilFjCkjEo8rdpdqPNerhy3XDi6ly8kAC/bin/darwin_ruby/lib/ruby/gems/3.1.0/gems/cocoapods-1.13.0/lib/cocoapods/installer/xcode/target_validator.rb:40:in `validate!'
    /Volumes/Work/s/w/ir/cache/ruby/.cipd/pkgs/0/pZTfjoFUwZNnilFjCkjEo8rdpdqPNerhy3XDi6ly8kAC/bin/darwin_ruby/lib/ruby/gems/3.1.0/gems/cocoapods-1.13.0/lib/cocoapods/installer.rb:666:in `validate_targets'
    /Volumes/Work/s/w/ir/cache/ruby/.cipd/pkgs/0/pZTfjoFUwZNnilFjCkjEo8rdpdqPNerhy3XDi6ly8kAC/bin/darwin_ruby/lib/ruby/gems/3.1.0/gems/cocoapods-1.13.0/lib/cocoapods/installer.rb:164:in `install!'
    /Volumes/Work/s/w/ir/cache/ruby/.cipd/pkgs/0/pZTfjoFUwZNnilFjCkjEo8rdpdqPNerhy3XDi6ly8kAC/bin/darwin_ruby/lib/ruby/gems/3.1.0/gems/cocoapods-1.13.0/lib/cocoapods/command/install.rb:52:in `run'
    /Volumes/Work/s/w/ir/cache/ruby/.cipd/pkgs/0/pZTfjoFUwZNnilFjCkjEo8rdpdqPNerhy3XDi6ly8kAC/bin/darwin_ruby/lib/ruby/gems/3.1.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
    /Volumes/Work/s/w/ir/cache/ruby/.cipd/pkgs/0/pZTfjoFUwZNnilFjCkjEo8rdpdqPNerhy3XDi6ly8kAC/bin/darwin_ruby/lib/ruby/gems/3.1.0/gems/cocoapods-1.13.0/lib/cocoapods/command.rb:52:in `run'
    /Volumes/Work/s/w/ir/cache/ruby/.cipd/pkgs/0/pZTfjoFUwZNnilFjCkjEo8rdpdqPNerhy3XDi6ly8kAC/bin/darwin_ruby/lib/ruby/gems/3.1.0/gems/cocoapods-1.13.0/bin/pod:55:in `<top (required)>'
    /Volumes/Work/s/w/ir/cache/ruby/bin/darwin_ruby/bin/pod:36:in `load'
    /Volumes/Work/s/w/ir/cache/ruby/bin/darwin_ruby/bin/pod:36:in `<main>'

Error output from CocoaPods:
↳
         Cloning into '/Volumes/Work/s/w/ir/x/t/d20240327-17982-ixo7ae'...
         Cloning into '/Volumes/Work/s/w/ir/x/t/d20240327-17982-k58n62'...

    [!] Automatically assigning platform `iOS` with version `12.0` on target `Runner` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

Error running pod install
```

Build: https://ci.chromium.org/ui/p/flutter/builders/prod/Mac_arm64%20ios_platform_tests_shard_1%20stable/1755/overview

**Comment by ditman** [comment]
This seems to be related to the update of the GoogleSignIn iOS SDK to version 7.1.0.

See:

* [GoogleSignIn iOS SDK Changelog](https://github.com/google/GoogleSignIn-iOS/releases)
* [Pod install fails on Codemagic but works locally](https://github.com/orgs/codemagic-ci-cd/discussions/2440)
* A related issue: https://github.com/react-native-google-signin/google-signin/issues/1263

I've posted a PR to pin our package to the 7.0.0 version, here:

* https://github.com/flutter/packages/pull/6430

(But we definitely want to update to 7.1.0, because it comes with Privacy Manifest)

**Comment by ditman** [comment]
A proper fix for this issue exists here:

* https://github.com/flutter/packages/pull/6404
